
###################################################################
###################### Configuration ##############################
###################################################################


type: "modem"


usbl_configuration:

  interface:

    connection_type: "tcp"

    tcp_address: "192.168.2.109"
    tcp_port: 9200

    device: /dev/ttyUSB0
    baudrate: 115200

  # Note: Transducer Source Level (0 = Maximum, 1 = -6dB, 2 = -12dB, 3 = -20dB)
  source_level: 3

  # Note: Source Control Level (0 = Cannot be changed by remote, 1 = Can be changed by remote)
  source_control: 1

  # Note: Gain (0 = Normal gain, 1 = Low gain (-20dB sensitivity) (short distance/testing))
  gain_level: 1

  # Note: Carrier Waveform ID (0-3) (0-1 for 2 devices, 2-2 and 3-3 for networking)
  carrier_waveform_id: 0

  # Note: Local Address (1-Highest Address)
  local_address: 1

  # Note: Remote Address (0-Highest Address) (0 can listen to all transmission but cannot send to any)
  remote_address: 2

  # Note: Highest Address (2,6,14,30,62,126,254)
  highest_address: 2

  # Note: Cluster Size (1-31) (Communication between moving objects should be <10. Stationary cases may be higher)
  cluster_size: 10

  # Note: Packet Time (500-1200ms) (Shorter packet time is recomended for challenging hydroacoustic channels)
  # MUST BE EQUAL FOR ALL COMMUNICATING DEVICES
  packet_time: 750

  # Note: Retry Count (0-255) The number of connection establishment retries
  retry_count: 3

  # Note: Retry Timeout (500-12000ms) The time interval the devices waits for a response to an acoustic connection establishment request
  # Retry Timeout should exced the two-way travel time of the maximum operating range plus 1 second
  retry_timeout: 4000

  #Note: Keep Online Count (0 = acoustic connection will remain online, 1-255 = number of message exchanges to keep the connection online)
  keep_online_count: 0

  #Note: Idle Timeout (0-3600s) The timeout before closing an idle acoustic connection
  idle_timeout: 120

  #Note: Channel Protocol ID (0-7) The data channel messages are exchanged on
  channel_protocol_id: 0

  #Note: Sound Speed (1300-1700m/s) The sound speed used to evaluate the range to the remote device
  sound_speed: 1500

modem_configuration:

  interface:

    connection_type: "tcp"

    tcp_address: "192.168.2.108"
    tcp_port: 9200

    device: /dev/ttyUSB0
    baudrate: 115200

  # Note: Transducer Source Level (0 = Maximum, 1 = -6dB, 2 = -12dB, 3 = -20dB)
  source_level: 0

  # Note: Source Control Level (0 = Cannot be changed by remote, 1 = Can be changed by remote)
  source_control: 1

  # Note: Gain (0 = Normal gain, 1 = Low gain (-20dB sensitivity) (short distance/testing))
  gain_level: 0

  # Note: Carrier Waveform ID (0-3) (0-1 for 2 devices, 2-2 and 3-3 for networking)
  carrier_waveform_id: 1

  # Note: Local Address (1-Highest Address)
  local_address: 2

  # Note: Remote Address (0-Highest Address) (0 can listen to all transmission but cannot send to any)
  remote_address: 1

  # Note: Highest Address (2,6,14,30,62,126,254)
  highest_address: 2

  # Note: Cluster Size (1-31) (Communication between moving objects should be <10. Stationary cases may be higher)
  cluster_size: 10

  # Note: Packet Time (500-1200ms) (Shorter packet time is recomended for challenging hydroacoustic channels)
  # MUST BE EQUAL FOR ALL COMMUNICATING DEVICES
  packet_time: 750

  # Note: Retry Count (0-255) The number of connection establishment retries
  retry_count: 1

  # Note: Retry Timeout (500-12000ms) The time interval the devices waits for a response to an acoustic connection establishment request
  # Retry Timeout should exced the two-way travel time of the maximum operating range plus 1 second
  retry_timeout: 1000

  #Note: Keep Online Count (0 = acoustic connection will remain online, 1-255 = number of message exchanges to keep the connection online)
  keep_online_count: 0

  #Note: Idle Timeout (0-3600s) The timeout before closing an idle acoustic connection
  idle_timeout: 3600

  #Note: Channel Protocol ID (0-7) The data channel messages are exchanged on
  channel_protocol_id: 0

  #Note: Sound Speed (1300-1700m/s) The sound speed used to evaluate the range to the remote device
  sound_speed: 1500

  







  








